docs/userguide/serialization.md

```diff
@@ -0,0 +1,390 @@
+
+While the above holds generally true for all classes that have converters in place,
+providing a few specific example may help to convey the concept:
+
+* Since {class}`Intervals <baybe.utils.interval.Interval>` can be created _implicitly_,
+    it is enough the specify their bound values directly:
+    ```python
+    from baybe.targets import NumericalTarget
+    from baybe.utils.interval import Interval
+
+    t1 = NumericalTarget(name="T", mode="MAX", bounds=Interval(0, 1))
+    t2 = NumericalTarget(name="T", mode="MAX", bounds=(0, 1))
+    t3 = NumericalTarget.from_json('{"name": "T", "mode": "MAX", "bounds": [0, 1]}')
+
+    assert t1 == t2 == t3
+    ```
+
+* Conversion to enums happens automatically whenever needed;
+    therefore, providing a raw string instead is sufficient:
+    ```python
+    from baybe.targets import NumericalTarget, TargetMode
+
+    t1 = NumericalTarget(name="T", mode=TargetMode.MAX)
+    t2 = NumericalTarget(name="T", mode="MAX")
+    t3 = NumericalTarget.from_json('{"name": "T", "mode": "MAX"}')
+
+    assert t1 == t2 == t3
+    ```
+
+### The `type` field
+Due to the leading design philosophy behind BayBE to provide its users easy access
+to a broad range of tools, you typically have the choice between several modelling 
+alternatives when building your objects.
+For example, when describing the degrees of freedom of your experimental campaign,
+you can chose from several different [parameter types](/userguide/parameters).
+
+While this offers great flexibility, it comes with a challenge for deserialization
+because you cannot know a priori which concrete object subclass is contained 
+in an incoming serialization string on the receiving end.
+Instead, you oftentimes need to be able to process the incoming string dynamically.
+
+For example, consider the following string, which perfectly mirrors the signatures of 
+both
+{meth}`CategoricalParameter <baybe.parameters.categorical.CategoricalParameter.__init__>` and 
+{meth}`TaskParameter <baybe.parameters.categorical.TaskParameter.__init__>`:
+```python
+parameter_str = """
+{
+    "name": "Setting",
+    "values": ["low", "high"]
+}
+"""
+```
+Unless you are aware of the specific purpose for which the string was created,
+calling one of the classes' constructors directly is impossible because you 
+simply do not know which one to chose.
+A similar situation arises with [nested objects](#NESTED_OBJECTS) because resorting to
+an explicit constructor call of a hand-selected subclass is only possible at the
+highest level of the hierarchy, whereas the inner object types would remain unspecified.
+
+The problem can be easily circumvented using an explicit subclass resolution 
+mechanism, i.e., by tagging the respective subclass in an additional `type` field that
+holds the class' name.
+This allows to deserialize the object from the corresponding base class instead,
+mirroring the flexibility of specifying subtypes to your configuration file:
```

Comment:
there needs o be an explicit mention here or via #comments to highlight that your using the abstract base class `Parameter` here, gets a bit lost in the code amount imo

Reply:
did both, good point